sdk/helper/aliasmetadata/alias_metadata.go

```diff
@@ -0,0 +1,195 @@
+	how these structs and fields are used in the AWS auth
+	method.
+
+	Or, check out its acceptance test in this package to
+	see its integration points.
+*/
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/strutil"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// Fields is for configuring a back-end's available
+// default and additional fields. These are used for
+// providing a verbose field description, and for parsing
+// user input.
+type Fields struct {
+	// The field name as it'll be reflected in the user-facing
+	// schema.
+	FieldName string
+
+	// Default is a list of the default fields that should
+	// be included if a user sends "default" in their list
+	// of desired fields. These fields should all have a
+	// low rate of change because each change can incur a
+	// write to storage.
+	Default []string
+
+	// AvailableToAdd is a list of fields not included by
+	// default, that the user may include.
+	AvailableToAdd []string
+}
+
+func (f *Fields) all() []string {
+	return append(f.Default, f.AvailableToAdd...)
+}
+
+// FieldSchema takes the default and additionally available
+// fields, and uses them to generate a verbose description
+// regarding how to use the "alias_metadata" field.
+func FieldSchema(fields *Fields) *framework.FieldSchema {
+	return &framework.FieldSchema{
+		Type:        framework.TypeCommaStringSlice,
+		Description: description(fields),
+		DisplayAttrs: &framework.DisplayAttributes{
+			Name:  fields.FieldName,
+			Value: "default,field1,field2",
+		},
+		Default: []string{"default"},
+	}
+}
+
+// NewHandler instantiates a Handler to be embedded in your config.
+func NewHandler(fields *Fields) Handler {
+	return &handler{
+		fields: fields,
+	}
+}
+
+// Handler is an interface for the helper methods you get on your
+// config when you embed the Handler.
+type Handler interface {
+	GetAliasMetadata() []string
+	ParseAliasMetadata(data *framework.FieldData) error
+	PopulateDesiredAliasMetadata(auth *logical.Auth, fieldValues map[string]string) error
+}
+
+type handler struct {
+	// AliasMetadata is an explicit list of all the user's configured
+	// fields that are being added to alias metadata. It will never
+	// include the "default" parameter, and instead includes the actual
+	// fields behind "default", if selected. If it has never been set,
+	// the pointer will be nil.
+	AliasMetadata []string `json:"alias_metadata"`
+
+	// fields is a list of the configured default and available
+	// fields. It's intentionally not jsonified.
+	fields *Fields
+}
+
+// GetAliasMetadata is intended to be used on config reads.
+// It gets an explicit list of all the user's configured
+// fields that are being added to alias metadata. It will never
+// include the "default" parameter, and instead includes the actual
+// fields behind "default", if selected.
+func (h *handler) GetAliasMetadata() []string {
```

Comment:
Totally! I did it that way to avoid a naming clash with the field named `AliasMetadata`. However, flipping over to custom JSON marshallers ought to fix the problem, will link the commit.

Reply:
https://github.com/hashicorp/vault/pull/8783/commits/f180042327b68ba3c6971e098478317348902aef